pkg/plugin/provision/spirehelm/spirehelm_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2024 Cofide Limited.
+// SPDX-License-Identifier: Apache-2.0
+
+package spirehelm
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	attestation_policy_proto "github.com/cofide/cofide-api-sdk/gen/go/proto/attestation_policy/v1alpha1"
+	provisionpb "github.com/cofide/cofide-api-sdk/gen/go/proto/provision_plugin/v1alpha1"
+	trust_zone_proto "github.com/cofide/cofide-api-sdk/gen/go/proto/trust_zone/v1alpha1"
+	"github.com/cofide/cofidectl/internal/pkg/config"
+	"github.com/cofide/cofidectl/internal/pkg/test/fixtures"
+	"github.com/cofide/cofidectl/pkg/plugin"
+	"github.com/cofide/cofidectl/pkg/plugin/local"
+	"github.com/cofide/cofidectl/pkg/plugin/provision"
+	"github.com/cofide/cofidectl/pkg/provider/helm"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSpireHelm_Deploy(t *testing.T) {
+	providerFactory := newFakeHelmSPIREProviderFactory()
+	spireHelm := NewSpireHelm(providerFactory)
+	ds := newFakeDataSource(t, defaultConfig())
+
+	statusCh, err := spireHelm.Deploy(context.Background(), ds, "fake-kube.cfg")
+	require.NoError(t, err, err)
+
+	statuses := collectStatuses(statusCh)
+	want := []*provisionpb.Status{
+		provision.StatusOk("Preparing", "Adding SPIRE Helm repo"),
+		provision.StatusDone("Prepared", "Added SPIRE Helm repo"),
+		provision.StatusOk("Installing", "Installing SPIRE CRDs for local1 in tz1"),
+		provision.StatusOk("Installing", "Installing SPIRE chart for local1 in tz1"),
+		provision.StatusDone("Installed", "Installation completed for local1 in tz1"),
+		provision.StatusOk("Installing", "Installing SPIRE CRDs for local2 in tz2"),
+		provision.StatusOk("Installing", "Installing SPIRE chart for local2 in tz2"),
+		provision.StatusDone("Installed", "Installation completed for local2 in tz2"),
+		provision.StatusOk("Waiting", "Waiting for SPIRE server pod and service for local1 in tz1"),
+		// FIXME: This attempts to create a real Kubernetes client and fails.
+		provision.StatusError(
+			"Waiting",
+			"Failed waiting for SPIRE server pod and service for local1 in tz1",
+			errors.New("load from file: open fake-kube.cfg: no such file or directory"),
+		),
```

Comment:
I guess we can mock this some point down the line, but failing here (as expected) seems fine to me

Reply:
Right - unsure what the appropriate interface is, but possibly something abstracting SPIRE communication. Or perhaps add it to Provider.